### ISSUE TITLE
Test failure blocking nix install

### ISSUE THREAD

**Issue by tdimiduk** [andrewthad/colonnade#7]
I am having a test failure blocking a nix install of colonnade:

Running 1 test suites...
Test suite test: RUNNING...

src/Colonnade.hs:38:1: error:
    Failed to load interface for ‘Colonnade.Encode’
    Use -v to see a list of the files searched for.

src/Colonnade.hs:44:1: error:
    Failed to load interface for ‘Colonnade.Encode’
    Use -v to see a list of the files searched for.
Test suite test: FAIL
Test suite logged to: dist/test/colonnade-1.1.0-test.log
0 of 1 test suites (0 of 1 test cases) passed.
builder for ‘/nix/store/ppsm7psg85np2g63h3sppf2qw4bcfnsx-colonnade-1.1.0.drv’ failed with exit code 1

I can install colonnade with cabal, and the nix install fails on two different machines (one running nixos and one running nixpkgs on ubuntu).

Please let me know if there is any more information I can provide to help debug this.

**Comment by mckeankylej** [comment]
I believe this is an issue with nix support of doctest Andrew do you agree?

**Comment by tdimiduk** [comment]
That seems likely. If @andrewthad agrees does one of you want to push this issue upstream, or should I?

Is it worth disabling test running for nix packages until that point?

**Comment by mckeankylej** [comment]
You can just disable the test for this package. There is a function in nixpkgs:

(import <nixpkgs> {}).haskell.lib.dontCheck 

that can turn off runing tests for a package. If you need a full example just ask.

**Comment by andrewthad** [comment]
I've never been able to get doctest to work with nix. I agree with Kyle's suggestion.

**Comment by tdimiduk** [comment]
Is that something that is easy for you to do, or do you want me to put
together a pull request against either nixpkgs or colonnade's nix code (not
positive where is the right place for it)? If so I could use the fuller
example from Kyle.

**Comment by mckeankylej** [comment]
Hey Tom,
Im glad you chose nix as your build tool, Its the best out there. I think before you start anything else you should read up on haskell and nix [here](http://nixos.org/nixpkgs/manual/#users-guide-to-the-haskell-infrastructure) and [here](https://github.com/Gabriel439/haskell-nix). With those guides in mind my following example will make much more sense!

```nix
# default.nix
let nixpkgs = import <nixpkgs> {};
    haskellPackages = nixpkgs.haskellPackages.override {
      overrides = self: super: {
        colonnade = nixpkgs.haskell.lib.dontCheck super.colonnade;
        your-package = self.callPackage ./path/to/your/package {};
      };
    };
    your-package = haskellPackages.your-package;
in
if nixpkgs.lib.inNixShell then
  your-package.env
else
  your-package
```
What this does is it creates a new haskell package set where colonnades tests do not get run and your package is added to the set. To use this you can either run `nix-shell default.nix` to drop you into a shell where you can run cabal commands with the correct deps or you can run `nix-build default.nix` to fully build the project. The later is preferred most of the time because cabal has a faster development cycle than nix-build. If you have any questions dont be afraid to ask.

**Comment by tdimiduk** [comment]
Thanks Kyle,

Nix is indeed awesome, I am slowly getting up to speed.

I have replaced my old shell.nix and now have colonnade working, so that is
great! Now we should just figure out the best way to get this into
colonnade or upstream nixpkgs so that the next person to come along doesn't
need to fight this problem.

**Comment by mckeankylej** [comment]
You can not turn off the test suite per package in nixpkgs. You can do it when you are overriding the package set in your own repo but they won't accept a pr that does that. This is very common place for people who know nix. I turn off test suites everyday when I use nix. The real solution is to fix doctest upstream!

**Comment by tdimiduk** [comment]
Ah, good to know. Thanks for teaching me how to disable the tests, it sounds like it will come in handy in the future. 

I am going to close this issue for now since it has a workaround and needs a fix upstream. Unfortunately I doubt my nix foo is strong enough to solve the upstream issue, but I'll keep it in mind.

**Comment by mckeankylej** [comment]
I'm not convinced it's anything to do with nix, probably a deficiency of doctest.

**Comment by mckeankylej** [comment]
Fixed with: fca7d72085e22ed0edd0a5d926bbcfb91ecda2e5
